main.rb

```diff
@@ -0,0 +1,308 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/AbcSize
+# rubocop:disable Metrics/MethodLength
+def my_each
+  return to_enum unless block_given?
+
+  i = 0
+  arr = to_a
+  while i <= arr.length - 1
+    yield (arr[i])
+    i += 1
+  end
+  self
+end
+
+def my_each_with_index
+  return to_enum unless block_given?
+
+  i = 0
+  arr = to_a
+  while i <= arr.length - 1
+    yield(arr[i], i)
+    i += 1
+  end
+  self
+end
+
+def my_all_option1(input, arr)
+  count = 0
+  arr.my_each do |x|
+    if input.is_a?(Integer)
+      count += 1 if x == input
+    elsif input.is_a?(Regexp)
+      count += 1 unless (x =~ input).nil?
+    elsif input.is_a?(Class)
+      count += 1 if x.is_a?(input)
+    end
+  end
+  count
+end
+
+def my_all?(input = nil)
+  arr = to_a
+  if block_given? == false && input.nil? == true
+    if block_given? == false && (arr.include?(false) == false && arr.include?(nil) == false)
+      true
+    else
+      false
+    end
+  elsif block_given? == false && !input.nil?
+    count = my_all_option1(input, arr)
+    count == arr.length
+  elsif block_given? == true
+    arr = to_a
+    count = 0
+    arr.my_each do |x|
+      count += 1 if yield(x) == true
+    end
+    count == arr.length
+  end
+end
+
+def my_any_option(input, arr)
```

Comment:
my_any_option is not part of the main method that is my_any?, I'use my_any_option to reduce the size of the main method so it do not support it  because I handle the array in the main and pass it to an array to my_any_option